utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/azure/AzureSchemaRegistryClientFactory.scala

```diff
@@ -0,0 +1,166 @@
+import com.azure.core.util.{Context, FluxUtil}
+import com.azure.data.schemaregistry.implementation.models.SchemasGetByIdResponse
+import com.azure.data.schemaregistry.models.{SchemaFormat, SchemaProperties, SchemaRegistrySchema}
+import com.azure.data.schemaregistry.{SchemaRegistryClientBuilder, SchemaRegistryVersion}
+import io.confluent.kafka.schemaregistry.ParsedSchema
+import io.confluent.kafka.schemaregistry.avro.AvroSchema
+import org.apache.commons.io.IOUtils
+import pl.touk.nussknacker.engine.kafka.SchemaRegistryClientKafkaConfig
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.azure.SchemaNameTopicMatchStrategy.FullSchemaNameDecomposed
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.azure.internal.{AzureConfigurationFactory, AzureHttpPipelineFactory, AzureTokenCredentialFactory, EnhancedSchemasImpl, SchemaRegistryJsonSerializer}
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.{SchemaId, SchemaNotFound, SchemaRegistryClientFactoryWithRegistration, SchemaRegistryClientWithRegistration, SchemaRegistryError, SchemaRegistryUnknownError, SchemaWithMetadata}
+import reactor.core.publisher.Mono
+
+import scala.compat.java8.FunctionConverters._
+import scala.jdk.CollectionConverters._
+import scala.util.control.NonFatal
+
+object AzureSchemaRegistryClientFactory extends SchemaRegistryClientFactoryWithRegistration {
+
+  override type SchemaRegistryClientT = AzureSchemaRegistryClient
+
+  override def create(config: SchemaRegistryClientKafkaConfig): SchemaRegistryClientT = {
+    new AzureSchemaRegistryClient(config)
+  }
+
+}
+
+class AzureSchemaRegistryClient(config: SchemaRegistryClientKafkaConfig) extends SchemaRegistryClientWithRegistration {
+
+  private val azureConfiguration = AzureConfigurationFactory.createFromKafkaProperties(config.kafkaProperties)
+  private val credential = AzureTokenCredentialFactory.createCredential(azureConfiguration)
+  private val httpPipeline = AzureHttpPipelineFactory.createPipeline(azureConfiguration, credential)
+  private val fullyQualifiedNamespace = config.kafkaProperties("schema.registry.url")
+  private val schemaGroup = config.kafkaProperties("schema.group")
+
+  private val schemaRegistryClient = new SchemaRegistryClientBuilder()
+    .fullyQualifiedNamespace(fullyQualifiedNamespace)
+    .credential(credential)
+    .buildClient()
+
+  // We need to create our own schemas service because some operations like schema listing are not exposed by default client
+  // or even its Schemas inner class. Others like listing of versions are implemented incorrectly (it has wrong json field name in model)
+  private val enhancedSchemasService = new EnhancedSchemasImpl(
+    fullyQualifiedNamespace,
+    SchemaRegistryVersion.getLatest.getVersion,
+    httpPipeline,
+    SchemaRegistryJsonSerializer)
+
+  override def getSchemaById(id: SchemaId): SchemaWithMetadata = {
+    toSchemaWithMetada(schemaRegistryClient.getSchema(id.asString))
```

Comment:
I think we need also `AvroUtils.adjustParsedSchema`  here

Reply:
I would rather like remove this at all: https://github.com/TouK/nussknacker/pull/3968 WDYT? 70 LOC less, everything works like expected, no synthetic entries, no worry to remember to adjust schema everywhere, I see only benefits